File: Server/src/Server.Application/ViewModels/ReservationViewModel.cs
namespace Server.Application.ViewModels;

public class ReservationViewModel
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public LibraryViewModel Library { get; set; } = null!;
    public string Status { get; set; } = null!;
    public DateTime ReservationEndDate { get; set; }
}

File: client/src/models/ReservationViewModel.ts
import { z } from 'zod';
import BookViewModel from './BookViewModel';

const LibraryViewModel = z.object({
  id: z.string(),
  name: z.string(),
});

const ReservationViewModel = z.object({
  id: z.string(),
  userId: z.string(),
  library: LibraryViewModel,
  status: z.string(),
  reservationEndDate: z.string(),
  books: z.array(BookViewModel).optional(),
});

export default ReservationViewModel;
export type ReservationViewModelType = z.infer<typeof ReservationViewModel>;

